src/components/ActionsPanel/ActionPanel.tsx

```diff
@@ -38,21 +40,35 @@ const HeadingText = styled.span`
   width: 250px;
 `;
 
-const filterUserActions = (actionTemplates: ActionTemplate[], state: StoreContextState) =>
+const filterUserActions = (actionTemplates: ActionTemplate[], filters: ActionFilters) =>
   actionTemplates.filter((action) => {
     const filter = action.filter;
     if (!filter) return true;
-    return filter(state);
+    return filter(filters);
   });
 
 export const ActionPanel = () => {
-  const storeContextState = useContext(StoreContext);
+  const { assets, uniClaims, ensOwnershipRecords, accounts, isMyCryptoMember } = useContext(
+    StoreContext
+  );
   const { userActions, updateUserAction, findUserAction } = useUserActions();
   const [currentAction, setCurrentAction] = useState<ActionTemplate | undefined>();
 
-  const relevantActions = useMemo(() => filterUserActions(actionTemplates, storeContextState), [
-    storeContextState
-  ]);
+  const relevantActions = useMemo(
+    () =>
+      pipe(
+        (a: ActionTemplate[]) =>
+          filterUserActions(a, {
+            assets,
+            uniClaims,
+            ensOwnershipRecords,
+            accounts,
+            isMyCryptoMember
+          }),
+        filter((a: ActionTemplate) => (a.time ? dateIsBetween(a.time.start, a.time.end) : true))
+      )(actionTemplates),
+    [assets, uniClaims, ensOwnershipRecords, accounts, isMyCryptoMember]
```

Comment:
Possibly this memo needs to be executed if the date changes? So the date between filter can be executed too.

Reply:
I prefer to wait for a app reload, it's pretty intensive to re-filter on every date change and I don't think it's useful